magento/magento2
# bugfix: fix cartFix rules incorrect discount ammount

- Add init totals into foreach rules loop to allow for total calculation after each rules apply on Discount.php

- remove logic for adding item discount to address base discount on Discount.php

- on CartFixed.php now use validator to calculate current valid items total discount for rule on proporcional discount calculation.

refs: #38694





### Description (*)


### Related Pull Requests


### Fixed Issues (if relevant)

1. Fixes magento/magento2#38694

### Manual testing scenarios (*)
1.     create multiple cartFix rules and apply them. 

3.     also, see the case reported in the issue to be sure. 


### Questions or comments
 i tested this on my project with my own integration tests. is there a simple way of seting up this github environment?
a one liner setup would be excelent if there is none but i dont have the time to set it up right now. 

### Contribution checklist (*)
